src/components/LabelField/LabelField.tsx

```diff
@@ -17,23 +18,28 @@ type FormWithLabels = {
 };
 
 export const LabelField = <T extends FormWithLabels>({ limit }: LabelFieldProps) => {
+  const { data: limits } = useTenantLimits();
   const { watch } = useFormContext<FormWithLabels>();
   const labels = watch('labels');
   const isEditor = hasRole(OrgRole.Editor);
-
-  let description = `Custom labels to be included with collected metrics and logs.`;
-
+  let description = '';
   if (limit) {
```

Comment:
I don't love this logic. Basically we're saying "some limits come from the api, and by not providing a limit prop we assume those are the ones you're looking for". We need that case for handling private probe label limits. Maybe just changing the name of the prop would help.

Reply:
I've changed it to something that makes me happier :)

Reply:
Nit: I'm not that precious about this but I'd extract that function to clean up the render method and reduce some responsibilities for each function.

```
 // in the render method
  const limit = getLimit(labelDestination, limits);
  const description = getDescription(labelDestination, limit, limits.maxAllowedLogLabels); 
```

```
// defined separately
function getLimit(labelDestination: LabelFieldProps['labelDestination'], limits: ListTenantLimitsResponse) {
  if (labelDestination === 'check') {
    return 3;
  }

  if (limits.maxAllowedMetricLabels) {
    return limits.maxAllowedMetricLabels;
  }

  return 10
}

function getDescription(labelDestination: LabelFieldProps['labelDestination'], limit: number, logLabelLimit: number) {
  if (labelDestination === 'check') {
    return `Custom labels to be included with collected metrics and logs. You can add up to ${limit}.`
  }

  return `Custom labels to be included with collected metrics and logs. You can add up to ${limit}. If you add more than ${logLabelLimit} labels, they will potentially not be used to index logs, and rather added as part of the log message.`
}
```